### ISSUE TITLE
After cutting the video, the video cannot be saved

### ISSUE THREAD

**Issue by yanghun0070** [maitrungduc1410/react-native-video-trim#16]
After cutting the video, the video cannot be saved.
The video cannot be played. An error appears saying

**Comment by maitrungduc1410** [comment]
Hi what error message is it?

Which platform are you facing the issue? Ios or android

Are you using real device?

**Comment by yanghun0070** [comment]
Occurs on android.
No error message occurs. However, if it appears as a blank screen as shown in the screenshot, a message will appear saying that there is no video when you click.
![Uploading 1.jpeg…](1)
![2](https://github.com/maitrungduc1410/react-native-video-trim/assets/22450400/2f361187-cf99-4db6-99f1-36eb6ccc4c58)

**Comment by yanghun0070** [comment]
I don’t think it can be saved properly after trimming.

**Comment by maitrungduc1410** [comment]
Is it possible for you to share me the sample video?

Also share me option you pass to showEditor

**Comment by yanghun0070** [comment]
옵션은 간단하다.```
isValidVideo(galleries[galleryIndex].uri).then((res) => console.log("isValidVideo:", res) );  showEditor(galleries[galleryIndex].uri, { 최대 기간: 30, }) .then((res) => console.log("res:" + res)) .catch(e) => console.log("e:" + e)); }}```

https://github.com/maitrungduc1410/react-native-video-trim/assets/22450400/d7364a51-2df5-41af-b15a-2a2fd27848aa

**Comment by yanghun0070** [comment]
There seems to be a file format error in iOS.
`onError {"message": "File is invalid", "name": "onError"}`

uri: "ph://D9958F6F-AC62-431F-A08A-6A991579F189"

**Comment by maitrungduc1410** [comment]
Okay, I'll check and get back to you soon

**Comment by maitrungduc1410** [comment]
I tried with your video on my Device, it's working fine.

As shown in my video, I did 2 samples, with same option for showEditor as yours.
- first sample: no change just trim, which should generate same video as original one
- second sample: choose a time range and trim

in the end of the video you can see that both samples are generated fine

https://github.com/maitrungduc1410/react-native-video-trim/assets/16630193/4aac3ef4-b48f-4f60-9169-803fcb79f895

**Comment by yanghun0070** [comment]
I'm getting the same error

https://github.com/maitrungduc1410/react-native-video-trim/assets/22450400/d099fc88-5dc3-4e0b-a91f-d69d407734b4

**Comment by maitrungduc1410** [comment]
Hi @yanghun0070 Can you try kill the photo app and reopen to see?

**Comment by yanghun0070** [comment]
After adding the code below, Android works normally.
```
  public static void saveVideoToGallery(ReactApplicationContext context, String videoFilePath) throws IOException {
    Log.w(TAG, "videoFilePath:" + videoFilePath);
    File videoFile = new File(videoFilePath);
    // Create the file if it doesn't exist
    if (!videoFile.exists()) {
      boolean isFileCreated = videoFile.createNewFile();
      Log.w(TAG, "File Created: " + isFileCreated);
    }

```

@maitrungduc1410 
The error below occurs in iOS. Do you know what causes it?

There seems to be a file format error in iOS.
onError {"message": "File is invalid", "name": "onError"}

uri: "ph://D9958F6F-AC62-431F-A08A-6A991579F189"

**Comment by maitrungduc1410** [comment]
On ios you need to run on real device, don't you?

**Comment by yanghun0070** [comment]
I think it was an image library issue.
I solved it by changing the ph:// path to the file path.

The added code is as follows.
```
    File videoFile = new File(videoFilePath);
    
    // Create the file if it doesn't exist
    if (!videoFile.exists()) {
      boolean isFileCreated = videoFile.createNewFile();
      Log.w(TAG, "File Created: " + isFileCreated);
    }
```

**Comment by maitrungduc1410** [comment]
fixed in 1.0.8
